Source/DataSources/CzmlDataSource.js

```diff
@@ -2150,11 +2152,28 @@ define([
         var promise = czml;
         var sourceUri = options.sourceUri;
 
+        // User specified credit
+        var credit = options.credit;
+        if (typeof credit === 'string') {
+            credit = new Credit(credit);
+        }
+        dataSource._credit = credit;
+
         // If the czml is a URL
         if (typeof czml === 'string' || (czml instanceof Resource)) {
             czml = Resource.createIfNeeded(czml);
             promise = czml.fetchJson();
             sourceUri = defaultValue(sourceUri, czml.clone());
+
+            // Add resource credits to our list of credits to display
+            var resourceCredits = dataSource._resourceCredits;
+            var credits = czml.credits;
+            if (defined(credits)) {
+                var length = credits.length;
+                for (var i = 0; i < length; i++) {
+                    resourceCredits.push(credits[i]);
```

Comment:
```suggestion
                    resourceCredits.push(new Credit(credits[i)]);
```

These definitely won't be `Credit` instances, right?

Reply:
They definitely are `Credit` objects. We iterate the `endpoint.attributions` and call `Credit.getIonCredit`, which returns a `Credit` instance.

Reply:
See `IonResource.getCreditsFromEndpoint`